literature_search.py

```diff
@@ -0,0 +1,243 @@
+#==============================================================================
+#=========================  Define the search criteria ========================
+
+from Bio import Entrez
+import numpy as np
+
+def search(query):
+    Entrez.email = 'your.email@example.com'
+    handle = Entrez.esearch(db='pubmed', 
+                            sort='most recent', 
+                            retmax='5000',
+                            retmode='xml', 
+                            datetype = 'pdat',
+                            reldate = 3, #only within n days from now
+#                             mindate = '2019/03/25',
+#                             maxdate = '2019/03/27', #for searching date range
+                            term=query)
+    results = Entrez.read(handle)
+    return results
+
+#search terms (can test string with Pubmed Advanced Search)
+search_results = search('(Biomech*[Title/Abstract] OR locomot*[Title/Abstract])')
+
+#==============================================================================
+#========================= Perform Search and Save Paper Titles ===============
+def fetch_details(id_list):
+    Entrez.email = 'your.email@example.com'
+    handle = Entrez.efetch(db='pubmed',
+                           retmode='xml',
+                           id=id_list)
+    results = Entrez.read(handle)
+    return results
+
+id_list = search_results['IdList']
+papers = fetch_details(id_list)
+print("")
+
+# Definitely could change these loops for speed.
+titles = [0 for i in enumerate(papers['PubmedArticle'])]
+keywords = ['' for i in enumerate(papers['PubmedArticle'])]
+authors = ['' for i in enumerate(papers['PubmedArticle'])]
+links = ['' for i in enumerate(papers['PubmedArticle'])]
+journals = ['' for i in enumerate(papers['PubmedArticle'])]
+abstracts = ['' for i in enumerate(papers['PubmedArticle'])]
+
+def clean_str(abs_string,stop):
+    translator = str.maketrans(string.punctuation, ' '*len(string.punctuation)) #map punctuation to space
+    abs_string = abs_string.translate(translator)
+    abs_string = abs_string.split()
+    abs_string = [word for word in abs_string if word not in stop]
+    abs_string = ' '.join(abs_string)
+    return abs_string
+
+# Make the Stop Words for string cleaning
+from nltk.corpus import stopwords
+import string
+stop = list(stopwords.words('english'))
+stop_c = [string.capwords(word) for word in stop]
+for word in stop_c:
+    stop.append(word)
+new_stop = ['StringElement','NlmCategory','Label','attributes','INTRODUCTION','METHODS','BACKGROUND','RESULTS','CONCLUSIONS']
+for item in new_stop:
+    stop.append(item)
+from nltk.corpus import stopwords
+
+for i, paper in enumerate(papers['PubmedArticle']):
+    titles[i] = clean_str(papers['PubmedArticle'][i]['MedlineCitation']['Article']['ArticleTitle'],stop)
+    abstracts[i] = clean_str(papers['PubmedArticle'][i]['MedlineCitation']['Article']['Abstract']['AbstractText'][0],stop)  
```

Comment:
So that paper doesn't have authors, abstract, a link. Thats just a broken paper which needs to be manually handeled. Its an error check that we need to add.